### ISSUE TITLE
deployed AppImage crashes on Fedora

### ISSUE THREAD

**Issue by kapitainsky** [probonopd/linuxdeployqt#400]
I build and deploy ma Qt app using Linux Ubuntu 16.04 LTS. When using Qt shipped with this OS all works fine. It is old and ugly so I try to build all using  newer Qt. But then I discovered that AppImages crash on Fedora when trying to open file dialogue. If run from terminal I can see "Segmentation fault (core dumped)"

I use linuxdeployqt release 6 - tried continues build - the same issue.

**Comment by kapitainsky** [comment]
Ubuntu 16.04 comes with Qt 5.5.1, I have tried 5.13.2, 5.13.1, 5.12.5 and  5.9.8. All works flawlessly (well despite openssl issues but I can deal with it) when build with 5.5.1. Other Qt work on Debian/Ubuntu but crash on Fedora (AppImage format)

**Comment by probonopd** [comment]
Please post the result of `env | grep QT` on the system that has the issue.

Then unset all `QT_` variables and launch your application again. Does it still crash?

**Comment by kapitainsky** [comment]
only one variable is set

`QT_IM_MODULE=ibus`

after `unset QT_IM_MODULE`

application crashes like before

**Comment by probonopd** [comment]
Then you need to use good old debugging techniques. Are you familiar with using `strace`?

**Comment by kapitainsky** [comment]
nope. But can learn:)

**Comment by probonopd** [comment]
```
./Your.AppImage --appimage-extract
strace -f ./squashfs-root/AppRun > log.txt 2>&1
```

Then reproduce the crash. Afterwards search through the log file for errors.
Compare with a similar log of a system that does not crash.

**Comment by kapitainsky** [comment]
I don't mind doing what is required but at first look it seems very gibberish for me..

[log.txt.zip](https://github.com/probonopd/linuxdeployqt/files/3799049/log.txt.zip)

Could you just look if maybe something obvious there?

**Comment by kapitainsky** [comment]
```
openat(AT_FDCWD, "/home/db/Downloads/squashfs-root/usr/plugins/platforms/../../lib/libgpg-error.so.0", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib64/libgpg-error.so.0", O_RDONLY|O_CLOEXEC) = 3
```
can it be some clue?

**Comment by probonopd** [comment]
No, it means that it was loaded from "/lib64/libgpg-error.so.0".

**Comment by kapitainsky** [comment]
what I can see that there is no clear error - it just gets terminated at some stage. I will compare with AppImage build with Qt 5.5.1 as it works. But not sure what clues it will give.. it will just not terminate..

**Comment by kapitainsky** [comment]
FYI - I solved the issue with another packaging tool.

https://github.com/linuxdeploy/linuxdeploy
with
https://github.com/linuxdeploy/linuxdeploy-plugin-qt

creates AppImage working without any issues on all systems I tested.

**Comment by probonopd** [comment]
This is _very_ interesting.
Can you upload an AppImage created with `linuxdeploy` and one with `linuxdeployqt`, so that we can compare? Thanks.

**Comment by kapitainsky** [comment]
If it help I will redo images no problem but I think I have identified the culprit

AppImage crashes on Fedora 31
`/usr/local/bin/linuxdeployqt-6-x86_64.AppImage AppDir/usr/share/applications/rclone-browser.desktop -no-translations -appimage -extra-plugins=iconengines,platformthemes/libqgtk3.so
`

AppImage works on Fedora 31
`/usr/local/bin/linuxdeployqt-6-x86_64.AppImage AppDir/usr/share/applications/rclone-browser.desktop -no-translations -appimage
`

**Comment by probonopd** [comment]
And how do the created AppImages differ?

**Comment by kapitainsky** [comment]
you want to see image deployed with and without `-extra-plugins=iconengines,platformthemes/libqgtk3.so` ?

**Comment by probonopd** [comment]
First, I'd be interested in a comparison of what is inside the two AppImages you mentioned in https://github.com/probonopd/linuxdeployqt/issues/400#issuecomment-549043706. Next, the comparison with and without the extra-plugins would also be interesting.

**Comment by kapitainsky** [comment]
ok - will do later today. I am interested in fixing it as this tool is very handy

**Comment by kapitainsky** [comment]
Here you are:

1. 
linuxdeployqt with extra plugins (crashes on Fedora 31)
`linuxdeployqt-6-x86_64.AppImage AppDir/usr/share/applications/rclone-browser.desktop -extra-plugins=iconengines,platformthemes/libqgtk3.so -appimage`

https://1drv.ms/u/s!Aq335pidOrBV3c9h7U8dYZmgw9C3ZA?e=6sKr6S


2.
linuxdeployqt without extra plugins (works OK)
`linuxdeployqt-6-x86_64.AppImage AppDir/usr/share/applications/rclone-browser.desktop -appimage`

https://1drv.ms/u/s!Aq335pidOrBV3c9grqxBhNLkjXpHfA?e=qgOSQY

3.
linuxdeploy with extra plugins (works OK)
```
linuxdeploy-x86_64.AppImage --appdir AppDir
linuxdeploy-plugin-qt-x86_64.AppImage  --appdir AppDir -p=iconengines,platformthemes/libqgtk3.so
/linuxdeploy-plugin-appimage-x86_64.AppImage --appdir=AppDir
```
https://1drv.ms/u/s!Aq335pidOrBV3c9fGhTgvZBZYuLpww?e=wpg9pz

**Comment by alemuntoni** [comment]
Hi, is there any news on this issue?
We are releasing AppImages of [MeshLab](https://github.com/cnr-isti-vclab/meshlab/releases) and we are using linuxdeployqt, but we have a [similar issue](https://github.com/cnr-isti-vclab/meshlab/issues/499#issuecomment-575958083) on Fedora.
I also tried myself the AppImage in a virtual machine, and I get a core dumped.

**Comment by probonopd** [comment]
@alemuntoni are you using any `-extra-plugins=...`? If yes, try without them.

**Comment by alemuntoni** [comment]
@probonopd nope, the only parameter is `-appimage`.
[This](https://github.com/cnr-isti-vclab/meshlab/blob/master/install/linux/linux_appimages.sh) is the script for deploying and generate the appimages, lines [67](https://github.com/cnr-isti-vclab/meshlab/blob/c6685ad27c8398b4e86273754731f66c8722fe2e/install/linux/linux_appimages.sh#L67) and [78](https://github.com/cnr-isti-vclab/meshlab/blob/c6685ad27c8398b4e86273754731f66c8722fe2e/install/linux/linux_appimages.sh#L78).

**Comment by probonopd** [comment]
@kapitainsky so it seems like linuxdeployqt deploying the dependencies for libqgtk3.so pulls in some Gtk stuff which is not safe to bundle. Workaround: do no use libqgtk3.so with linuxdeployqt.

**Comment by probonopd** [comment]
* https://github.com/probonopd/linuxdeployqt/issues/236
